TST, MAINT: array API GPU test fails in `test_cubature.py`

Probably slipped through after gh-21657. There are 30 new fails on `main`, but they only fall into two categories (really just 1 category with slightly different errors), so I'll just paste the representative CuPy-only examples below the fold.

<details>

```
________________________________________________ TestCubatureProblems.test_infinite_limits_and_break_points[problem1-cupy-genz-malik-1e-05-0.0001] ________________________________________________
[gw5] linux -- Python 3.11.2 /home/treddy/python_venvs/py_311_scipy_dev/bin/python
scipy/integrate/tests/test_cubature.py:1177: in test_infinite_limits_and_break_points
    res = cubature(
        a          = array([-inf,   0.])
        atol       = 1e-05
        b          = array([inf,  1.])
        exact      = array(1.50596981)
        f          = <function TestCubatureProblems.<lambda> at 0x7f8bb9225940>
        ndim       = 2
        points     = [array([0. , 0.5])]
        problem    = (<function TestCubatureProblems.<lambda> at 0x7f8bb9225940>, 1.5059698117208216, [-inf, 0], [inf, 1], [[0, 0.5]])
        rtol       = 0.0001
        rule       = 'genz-malik'
        self       = <scipy.integrate.tests.test_cubature.TestCubatureProblems object at 0x7f8bb908cb90>
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
scipy/integrate/_cubature.py:405: in cubature
    est_k = rule.estimate(f, a_k, b_k, args)
        a          = array([-1.,  0.])
        a_flipped  = array([-inf,   0.])
        a_k        = array([-1.,  0.])
        args       = (<module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>,)
        atol       = 1e-05
        b          = array([1., 1.])
        b_flipped  = array([inf,  1.])
        b_k        = array([0., 1.])
        err        = 0.0
        est        = 0.0
        f          = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8b638ce710>
        initial_regions = [(array([-1.,  0.]), array([0., 1.])), (array([0., 0.]), array([1., 1.]))]
        max_subdivisions = 10000
        ndim       = 2
        points     = [array([inf, inf]), array([0., 0.])]
        regions    = []
        result_dtype = dtype('float64')
        rtol       = 0.0001
        rule       = <scipy.integrate._rules._genz_malik.GenzMalikCubature object at 0x7f8b638ce150>
        sign       = array(1.)
        workers    = 1
        xp         = <module 'scipy._lib.array_api_compat.cupy' from '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/_lib/array_api_compat/cupy/__init__.py'>
scipy/integrate/_rules/_base.py:242: in estimate
    return _apply_fixed_rule(f, a, b, nodes, weights, args, self.xp)
        a          = array([-1.,  0.])
        args       = (<module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>,)
        b          = array([0., 1.])
        f          = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8b638ce710>
        nodes      = array([[ 0.        ,  0.        ],
       [ 0.        ,  0.35856858],
       [ 0.35856858,  0.        ],
       [-0.35... 0.6882472 ],
       [ 0.6882472 , -0.6882472 ],
       [-0.6882472 ,  0.6882472 ],
       [-0.6882472 , -0.6882472 ]])
        self       = <scipy.integrate._rules._genz_malik.GenzMalikCubature object at 0x7f8b638ce150>
        weights    = array([-0.77549154,  0.5974699 ,  0.5974699 ,  0.5974699 ,  0.5974699 ,
        0.20728547,  0.20728547,  0.20728547, ...  0.04064421,
        0.04064421,  0.04064421,  0.04064421,  0.3484733 ,  0.3484733 ,
        0.3484733 ,  0.3484733 ])
scipy/integrate/_rules/_base.py:510: in _apply_fixed_rule
    f_nodes = f(nodes, *args)
        a          = array([-1.,  0.])
        a_ndim     = 2
        args       = (<module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>,)
        b          = array([0., 1.])
        b_ndim     = 2
        f          = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8b638ce710>
        lengths    = array([1., 1.])
        nodes      = array([[-0.5       ,  0.5       ],
       [-0.5       ,  0.67928429],
       [-0.32071571,  0.5       ],
       [-0.67... 0.8441236 ],
       [-0.1558764 ,  0.1558764 ],
       [-0.8441236 ,  0.8441236 ],
       [-0.8441236 ,  0.1558764 ]])
        orig_nodes = array([[ 0.        ,  0.        ],
       [ 0.        ,  0.35856858],
       [ 0.35856858,  0.        ],
       [-0.35... 0.6882472 ],
       [ 0.6882472 , -0.6882472 ],
       [-0.6882472 ,  0.6882472 ],
       [-0.6882472 , -0.6882472 ]])
        orig_weights = array([-0.77549154,  0.5974699 ,  0.5974699 ,  0.5974699 ,  0.5974699 ,
        0.20728547,  0.20728547,  0.20728547, ...  0.04064421,
        0.04064421,  0.04064421,  0.04064421,  0.3484733 ,  0.3484733 ,
        0.3484733 ,  0.3484733 ])
        result_dtype = dtype('float64')
        rule_ndim  = 2
        weight_scale_factor = array(0.25)
        weights    = array([-0.19387289,  0.14936747,  0.14936747,  0.14936747,  0.14936747,
        0.05182137,  0.05182137,  0.05182137, ...  0.01016105,
        0.01016105,  0.01016105,  0.01016105,  0.08711833,  0.08711833,
        0.08711833,  0.08711833])
        xp         = <module 'scipy._lib.array_api_compat.cupy' from '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/_lib/array_api_compat/cupy/__init__.py'>
scipy/integrate/_cubature.py:721: in __call__
    self._xp.reshape(
        args       = (<module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>,)
        double_inf_mask = array([[ True, False],
       [ True, False],
       [ True, False],
       [ True, False],
       [ True, False],
   ...e],
       [ True, False],
       [ True, False],
       [ True, False],
       [ True, False],
       [ True, False]])
        kwargs     = {}
        npoints    = 17
        self       = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8b638ce710>
        semi_inf_mask = array([[False, False],
       [False, False],
       [False, False],
       [False, False],
       [False, False],
   ...e],
       [False, False],
       [False, False],
       [False, False],
       [False, False],
       [False, False]])
        start      = array([], dtype=float64)
        t          = array([[-0.5       ,  0.5       ],
       [-0.5       ,  0.67928429],
       [-0.32071571,  0.5       ],
       [-0.67... 0.8441236 ],
       [-0.1558764 ,  0.1558764 ],
       [-0.8441236 ,  0.8441236 ],
       [-0.8441236 ,  0.1558764 ]])
        x          = array([[-1.00000000e+00,  5.00000000e-01],
       [-1.00000000e+00,  6.79284291e-01],
       [-2.11802626e+00,  5.0000....41533937e+00,  1.55876399e-01],
       [-1.84660634e-01,  8.44123601e-01],
       [-1.84660634e-01,  1.55876399e-01]])
scipy/_lib/array_api_compat/_internal.py:28: in wrapped_f
    return f(*args, xp=xp, **kwargs)
        args       = (array([2.50000000e-01, 2.50000000e-01, 1.02858566e-01, 4.61427149e-01,
       2.50000000e-01, 2.50000000e-01, 6.58350...75e-04,
       9.49341649e-01, 2.42974518e-02, 2.42974518e-02, 7.12544653e-01,
       7.12544653e-01]), (-1, array(1)))
        f          = <function reshape at 0x7f8bcfc9eca0>
        kwargs     = {}
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
scipy/_lib/array_api_compat/common/_aliases.py:343: in reshape
    return xp.reshape(x, shape, **kwargs)
        copy       = None
        kwargs     = {}
        shape      = (-1, array(1))
        x          = array([2.50000000e-01, 2.50000000e-01, 1.02858566e-01, 4.61427149e-01,
       2.50000000e-01, 2.50000000e-01, 6.583509...649e-01, 6.58350975e-04,
       9.49341649e-01, 2.42974518e-02, 2.42974518e-02, 7.12544653e-01,
       7.12544653e-01])
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_manipulation/shape.py:55: in reshape
    return a.reshape(newshape, order=order)
        a          = array([2.50000000e-01, 2.50000000e-01, 1.02858566e-01, 4.61427149e-01,
       2.50000000e-01, 2.50000000e-01, 6.583509...649e-01, 6.58350975e-04,
       9.49341649e-01, 2.42974518e-02, 2.42974518e-02, 7.12544653e-01,
       7.12544653e-01])
        newshape   = (-1, array(1))
        order      = 'C'
cupy/_core/core.pyx:758: in cupy._core.core._ndarray_base.reshape
    ???
cupy/_core/_routines_manipulation.pyx:82: in cupy._core._routines_manipulation._ndarray_reshape
    ???
stringsource:48: in vector.from_py.__pyx_convert_vector_from_py_Py_ssize_t
    ???
E   TypeError: 'ndarray' object cannot be interpreted as an integer
__________________________________________________________ TestTransformations.test_infinite_limits_maintains_points[a0-b0-points0-cupy] __________________________________________________________
[gw5] linux -- Python 3.11.2 /home/treddy/python_venvs/py_311_scipy_dev/bin/python
scipy/integrate/tests/test_cubature.py:1373: in test_infinite_limits_maintains_points
    f_transformed(transformed_point)
        a          = [0, 1, -inf]
        b          = [1, inf, inf]
        f_transformed = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8ba69ea190>
        point      = array([1., 1., 1.])
        points     = [array([1., 1., 1.]), array([ 0.5, 10. , 10. ])]
        self       = <scipy.integrate.tests.test_cubature.TestTransformations object at 0x7f8bb90b2690>
        transformed_point = array([[1. , 1. , 0.5]])
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
        xp_compat  = <module 'scipy._lib.array_api_compat.cupy' from '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/_lib/array_api_compat/cupy/__init__.py'>
scipy/integrate/_cubature.py:721: in __call__
    self._xp.reshape(
        args       = ()
        double_inf_mask = array([[False, False,  True]])
        kwargs     = {}
        npoints    = 1
        self       = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8ba69ea190>
        semi_inf_mask = array([[False,  True, False]])
        start      = array([1.])
        t          = array([[1. , 1. , 0.5]])
        x          = array([[1., 1., 1.]])
scipy/_lib/array_api_compat/_internal.py:28: in wrapped_f
    return f(*args, xp=xp, **kwargs)
        args       = (array([1.  , 0.25]), (-1, array(2)))
        f          = <function reshape at 0x7f8bcfc9eca0>
        kwargs     = {}
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
scipy/_lib/array_api_compat/common/_aliases.py:343: in reshape
    return xp.reshape(x, shape, **kwargs)
        copy       = None
        kwargs     = {}
        shape      = (-1, array(2))
        x          = array([1.  , 0.25])
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_manipulation/shape.py:55: in reshape
    return a.reshape(newshape, order=order)
        a          = array([1.  , 0.25])
        newshape   = (-1, array(2))
        order      = 'C'
cupy/_core/core.pyx:758: in cupy._core.core._ndarray_base.reshape
    ???
cupy/_core/_routines_manipulation.pyx:82: in cupy._core._routines_manipulation._ndarray_reshape
    ???
stringsource:48: in vector.from_py.__pyx_convert_vector_from_py_Py_ssize_t
    ???
E   TypeError: 'ndarray' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:
scipy/integrate/tests/test_cubature.py:1372: in test_infinite_limits_maintains_points
    with pytest.raises(Exception, match="called with a problematic point"):
E   AssertionError: Regex pattern did not match.
E    Regex: 'called with a problematic point'
E    Input: "'ndarray' object cannot be interpreted as an integer"
        a          = [0, 1, -inf]
        b          = [1, inf, inf]
        f_transformed = <scipy.integrate._cubature._InfiniteLimitsTransform object at 0x7f8ba69ea190>
        point      = array([1., 1., 1.])
        points     = [array([1., 1., 1.]), array([ 0.5, 10. , 10. ])]
        self       = <scipy.integrate.tests.test_cubature.TestTransformations object at 0x7f8bb90b2690>
        transformed_point = array([[1. , 1. , 0.5]])
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
        xp_compat  = <module 'scipy._lib.array_api_compat.cupy' from '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/_lib/array_api_compat/cupy/__init__.py'>
```

</details>

--- a/scipy/integrate/_cubature.py
+++ b/scipy/integrate/_cubature.py
@@ -633,7 +633,7 @@ class _InfiniteLimitsTransform(_VariableTransform):
 
         self._num_inf = self._xp.sum(
             self._xp.astype(self._double_inf_pos | self._semi_inf_pos, self._xp.int64),
-        )
+        ).__int__()
 
     @property
     def transformed_limits(self):

